software/contrib/hamlet.py

```diff
@@ -0,0 +1,359 @@
+* Two channels of gate and random stepped CV with sparsity control. As
+  the sparsity is turned up, notes will drop out of the pattern.
+
+digital_in: clock in
+analog_in: sparsity CV
+
+knob_1: sparsity
+knob_2: select pre-loaded drum pattern
+
+button_1: Short Press: Play previous CV Pattern. Long Press: Change CV track length multiplier
+button_2: Short Press: Generate a new random cv pattern for Tracks 1 and 2. Long Press: Cycle through analogue input modes
+output_1: trigger 1 / Bass Drum
+output_2: trigger 2 / Hi-Hat
+output_3: gates for track 1 
+output_4: gates for track 2
+output_5: randomly generated track 2 CV (cycled by pushing button 2)
+output_6: randomly generated track 1 CV (cycled by pushing button 2)
+
+'''
+
+class Hamlet(EuroPiScript):
+    def __init__(self):
+        self.bd = cv1
+        self.hh = cv2
+        self.gate_1 = cv3
+        self.cv_1 = cv6
+        self.gate_2 = cv4
+        self.cv_2 = cv5
+
+        # Overclock the Pico for improved performance.
+        machine.freq(250_000_000)
+
+        # Initialize sequencer pattern arrays   
+        p = pattern()     
+        self.BD=p.BD
+        self.HH=p.HH
+
+        # Initialize variables
+        self.drum_step = 0
+        self.track_step = 0
+
+        self.trigger_duration_ms = 50
+        self.clock_step = 0
+        self.pattern = 0
+        self.minAnalogInputVoltage = 0.9
+        self.randomness = 0
+        self.sparsity = 0
+        self.analogInputMode = 1 # 1: Randomness, 2: Pattern, 3: CV Pattern
+        self.CvPattern = 0
+        
+        # Generate random CV for cv4-6
+        self.track_1 = []
+        self.track_2 = []
+
+        self.track_multiplier = 1 # multiplies CV pattern length so we can have 16, 32, 64
+        
+        self.generateNewRandomCVPattern()
+
+        # Triggered when button 2 is released.
+        # Short press: Generate random CV for Tracks 1 and 2
+        # Long press: Change operating mode
+        @b2.handler_falling
+        def b2Pressed():
+            
+            if ticks_diff(ticks_ms(), b2.last_pressed()) >  300:
+                if self.analogInputMode < 3:
+                    self.analogInputMode += 1
+                else:
+                    self.analogInputMode = 1
+            else:
+                # Move to next cv pattern if one already exists, otherwise create a new one
+                self.CvPattern += 1
+                if self.CvPattern == len(self.track_1):
+                    self.generateNewRandomCVPattern()
+            
+        # Triggered when button 1 is released
+        # Short press: Play previous CV pattern for cv4-6
+        # Long press: Not used
+        @b1.handler_falling
+        def b1Pressed():
+            if ticks_diff(ticks_ms(), b1.last_pressed()) >  300:
+                if self.track_multiplier >=4:
+                    self.track_multiplier = 1
+                else:
+                    self.track_multiplier = self.track_multiplier * 2
```

Comment:
Optional: Simplify with an assignment operator.
```suggestion
                    self.track_multiplier *= 2
```
https://docs.python.org/3/reference/simple_stmts.html#augmented-assignment-statements